src/query/cost/cost.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+package cost
+
+import (
+	"fmt"
+
+	"github.com/m3db/m3/src/x/cost"
+	"github.com/m3db/m3x/instrument"
+
+	"github.com/uber-go/tally"
+)
+
+// NB (amains) type alias here allows us the same visibility restrictions as defining an interface
+// (PerQueryEnforcerOpts isn't directly constructable), but without the unnecessary boilerplate.
+
+// PerQueryEnforcerOpts configures a PerQueryEnforcer.
+type PerQueryEnforcerOpts struct {
+	valueBuckets   tally.Buckets
+	instrumentOpts instrument.Options
+}
+
+// NewPerQueryEnforcerOpts constructs a default PerQueryEnforcerOpts
+func NewPerQueryEnforcerOpts(instrumentOps instrument.Options) *PerQueryEnforcerOpts {
+	return &PerQueryEnforcerOpts{
+		instrumentOpts: instrumentOps,
+		valueBuckets:   tally.MustMakeExponentialValueBuckets(1, 2, 20),
```

Comment:
> Shouldn't this come from config?

Normally it will; these are defaults. 

> Any chance can convert this to use tally.MakeExponentialValueBuckets and return the error if it fails rather than panicing?

Most times I'd agree with you on returning the error, but since these are defaults and will remain constant, any error in them is a programming error, and therefore better to panic on (the caller can't reasonably handle it). I'll add a test to ensure that it's valid though.